File: solutions.py
#import regular expressions and system functionality
import re
import sys

'''
numbers = list()
sum = 0

for i in xrange(1000):
    if i % 3 == 0 or i % 5 == 0 :
       numbers.append(i)
       sum += i

#print numbers
print sum

prev = 0
curr = 1
sum = 0

while curr < 4000000:
    if ( (prev + curr) % 2 == 0):
        sum += (prev + curr)
    temp = prev
    prev = curr
    curr = temp + curr
    #print curr

print "sum equals: %d" %sum


#Problem 4
#find largest palndrome product from two 3 digit numbers
curr = 999
answer = 0
while curr > 99:
    for x in range (1000, 100, -1):
        product = curr * x
        if (str(product) == str(product)[::-1] and product > answer):
            answer = product
    curr -= 1
print answer


#Problem 5
i = 21
solved = 0
while (not(solved)):
    for x in range (21, 1, -1):
        if (i % x != 0):
            solved = 0
            i += 1
            break
        else:
            solved = 1
print i


print type(3.2)

if (type(3.2) is float):
    print "it's a float!"

x = 2**5+1
print x

print "now strings"
string1 = "this is"
string2 = " python speaking"
print string1 + string2

print "now user input"
input = raw_input('Please enter text here to have repeated to you:\n')
print input
'''

print "Exercise 2.3"
hours = raw_input("Enter the number of hours worked:   ")
rate = raw_input("Enter the pay rate of the worker:   ")
print "Pay:   " + str(35*2.75) + "\n"

print "done"

File: solutions.rb
#!/usr/local/bin/ruby

array = []
sum = 0

##############
# Problem 1
##############
for i in 0..999
    if ((i % 3) == 0 || (i % 5) == 0) then
        array.push(i)
        sum = sum + i
    end
end

puts sum

##############
# Problem 2
##############
sum = 0
prev = 0
curr = 1
while curr < 4000000
    if ( (prev + curr) % 2 == 0)
        sum += (prev + curr)
    end
    temp = prev
    prev = curr
    curr = temp + curr
    #puts curr
end

puts "answer for problem 2 is: " + sum.to_s

=begin
##############
# Problem 3
##############
# largest prime factor of 600851475143
array = []
for i in 1...600851475143
    if (600851475143 % i == 0)
    #if (839 % i == 0)
        array.push(i)
    end
end
puts array
=end

###############
# Problem 4
###############
answer = 0
for i in 999.downto(99)
    for j in 999.downto(99)
        product = i*j
        if (product.to_s == product.to_s.reverse && product > answer)
            answer = product
        end
    end
end
puts answer
